src/NServiceBus.Core/Pipeline/MainPipelineExecutor.cs

```diff
@@ -28,10 +29,11 @@ public async Task Invoke(MessageContext messageContext)
                 var rootContext = new RootContext(childScope.ServiceProvider, messageOperations, pipelineCache);
                 rootContext.Extensions.Merge(messageContext.Extensions);
 
-                var transportReceiveContext = new TransportReceiveContext(message, messageContext.TransportTransaction, messageContext.ReceiveCancellationTokenSource, rootContext);
+                var transportReceiveContext = new TransportReceiveContext(message, messageContext.TransportTransaction, rootContext);
 
                 try
                 {
+                    //TODO: Pass the token to the pipeline once Davids PR is in
                     await receivePipeline.Invoke(transportReceiveContext).ConfigureAwait(false);
```

Comment:
the transport will deliver a `MessageContext` and a `CancallationToken` as separate parameters to us. I guess we could merge them but since this is the low-level seam I think it's better to follow MS best practices?